### ISSUE TITLE
XLA tf.random.stateless_uniform produces different values from eager and graph for the same seed

### ISSUE THREAD

**Issue by laolvfan** [tensorflow/tensorflow#121243]
### Issue type

Bug

### Have you reproduced this with the latest TensorFlow nightly?

Yes. Reproduced with `tf-nightly 2.22.0-dev20260531`.

### Standalone code to reproduce the issue

```python
import tensorflow as tf

print("TensorFlow:", tf.__version__)

seed = [1, 2]
shape = [2, 2]

print("eager:", tf.random.stateless_uniform(shape, seed=seed, dtype=tf.float32).numpy())

@tf.function
def graph_f():
    return tf.random.stateless_uniform(shape, seed=seed, dtype=tf.float32)

@tf.function(jit_compile=True)
def xla_f():
    return tf.random.stateless_uniform(shape, seed=seed, dtype=tf.float32)

print("graph:", graph_f().numpy())
print("xla:", xla_f().numpy())
```

### Observed output

```text
TensorFlow: 2.22.0-dev20260531

eager:
[[0.8440604  0.19204533]
 [0.9962232  0.1603874 ]]

graph:
[[0.8440604  0.19204533]
 [0.9962232  0.1603874 ]]

xla:
[[0.25750792 0.4046719 ]
 [0.9218676  0.08625066]]
```

### Expected behavior

`tf.random.stateless_uniform` should produce the same values for the same seed and shape under eager, non-XLA graph, and XLA execution on the same CPU device.

The API documentation says this is the stateless version of `tf.random.uniform`: running twice with the same seeds and shapes produces the same pseudorandom numbers, and the output is consistent across multiple runs on the same hardware and between CPU and GPU. This reproducer runs on the same host CPU but XLA produces different values from eager and non-XLA graph execution.

### Additional details

This was found while minimizing a WhiteFox TFXLA differential test case. The original generated model reduced to a single `tf.random.stateless_uniform([2, 2], seed=[1, 2], dtype=tf.float32)` call.

**Comment by mansiverma897993** [comment]
Is this issue resloved ?

**Comment by laolvfan** [comment]
As far as I can tell, not yet. The issue is still open.

The related PR (#121252) seems to clarify the documentation for `tf.random.stateless_*` under XLA, but it does not change the runtime behavior. So the eager/non-XLA graph vs XLA output difference appears to remain.

**Comment by laolvfan** [comment]
I found another minimized reproducer from a WhiteFox TFXLA differential test. It appears to be the same root cause as this issue, but with `tf.random.stateless_normal` instead of `tf.random.stateless_uniform`.

The originally generated program used `tf.keras.initializers.RandomNormal` inside `Model.call()`, but minimization shows the Keras/model parts are not necessary. A single stateless RNG op is enough:

```python
import tensorflow as tf

print("tf", tf.__version__)

seed = tf.constant([1, 2], tf.int32)
shape = [2, 2]

print("eager")
print(tf.random.stateless_normal(shape, seed=seed, dtype=tf.float32).numpy())

@tf.function
def graph_f():
    return tf.random.stateless_normal(shape, seed=seed, dtype=tf.float32)

@tf.function(jit_compile=True)
def xla_f():
    return tf.random.stateless_normal(shape, seed=seed, dtype=tf.float32)

print("graph")
print(graph_f().numpy())
print("xla")
print(xla_f().numpy())
```

Observed with `tf-nightly 2.22.0-dev20260531` on CPU:

```text
tf 2.22.0-dev20260531
eager
[[0.5441101  0.20738031]
 [0.07356432 0.04643455]]
graph
[[0.5441101  0.20738031]
 [0.07356432 0.04643455]]
xla
[[-0.77657187  0.6938005 ]
 [ 1.4527029   1.152388  ]]
```

Replacing the random tensor with a fixed `tf.constant` makes eager, non-XLA graph, and XLA agree, so the mismatch appears to be localized to stateless RNG lowering under XLA.

Since this affects `stateless_normal` as well as `stateless_uniform`, it looks like a family-wide runtime divergence rather than a documentation issue for a single API. This seems consistent with PR #121416, which proposes a runtime fix (removing the double-scrambling of the Philox key) covering `stateless_uniform`, `stateless_normal`, and other stateless RNG ops.
